### DISCUSSION TITLE
Migration from skoruba / IdentityServer4.Admin

### DISCUSSION THREAD

**Discussion by darshanllakhwani** [skoruba/Duende.IdentityServer.Admin#120] [Q&A]
As i am using old version of IdentityServer4Admin and now i would to like to switch to latest version of this project.

my question is -

1.is it required any license ?
2.How we can migrate the data from old to new ?
3How we can migrate the code from old version to new version (as we did some customisation over the code )?

Looking for help.

**Comment by skoruba** [comment]
Hey, you need license key for Duende Identity Server according to these requirements: https://duendesoftware.com/products/identityserver#pricing

You can apply all new database migrations - but try it on development and be sure that it works good for you.

What type of changes did you do in the code? Do you use AdminUI package or did you fork whole solution?

**Reply by darshanllakhwani** [reply]
Yes we have forked the all solution and we did the changes mostly in HTML example - (some css changes and branding  changes)  and few changes in api side  examples - (security changes for password )

**Reply by skoruba** [reply]
I think you can do it absolutely same way in the Duende Admin IS as well, but unfortunately - you forked and there is no way how to easily merge it with new version.
